src/streamer.ts

```diff
@@ -0,0 +1,36 @@
+import { S3Client } from '@aws-sdk/client-s3';
+import { listBlocks, fetchStreamerMessage } from './s3fetchers';
+import { LakeConfig, BlockHeight, StreamerMessage } from './types';
+
+export async function startStream(
+  config: LakeConfig,
+  onStreamerMessageReceived: (data: StreamerMessage) => Promise<void>
+) {
+  const s3Client = new S3Client({ region: config.s3RegionName });
+
+  let lastProcessedBlockHash: string;
+  let startFromBlockHeight = config.startBlockHeight;
+  let queue: Promise<void>[] = [];
+
+  while (true) {
```

Comment:
Shouldn't we let user stop streaming?
I believe we can reuse [AbortController interface](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) (though I'm not aware if it's a popular way to go in node.js universe)

Reply:
CTRL+C is stopping the thing. In the real-world use case, a user wants the stream to be ongoing

Reply:
@khorolets Well, it is better to have an option to stop the process programmatically whenever it is possible (not only by aborting the whole app). Imagine I want to implement an indexer that fills the gaps, and so I will start the indexer at certain point and then after I reached certain block I want to stop this indexer and start a new one at the next gap.

Reply:
@frol it sounds like a feature. We don't have such a feature on the Rust version which is the main one, so I don't see any reason to bring it here at least for now.

Reply:
@khorolets Please, extract this into a separate issue to address it later.

Reply:
For the future reference: #2